packages/admin-ui/src/masked-input/use-formatter.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo } from 'react'
+
+const MASK_INPUT_SYMBOL = '_'
+
+export interface UseFormatterProps {
+  mask: string
+  accept: RegExp
+}
+
+export function useFormatter(props: UseFormatterProps) {
+  const { mask, accept } = props
+  const formatter = useMemo(
+    () =>
+      createFormatter({
+        mask,
+        accept,
+      }),
+    [mask, accept]
+  )
+
+  return formatter
+}
+
+function createFormatter(props: UseFormatterProps) {
+  const { mask, accept } = props
+
+  function formatter(value: string) {
+    return value
+      .split('')
+      .map((char, i) => {
+        accept.lastIndex = 0
+
+        if (i > mask.length - 1) {
+          return ''
+        }
+
+        const maskChar = mask[i]
+        const nextMaskChar = mask[i + 1]
+
+        const acceptedChar = accept.test(char) ? char : ''
+        const formattedChar =
```

Comment:
```suggestion
       formatChar(char, maskChar, accept) {
           const acceptedChar = accept.test(char)
           
           if (!acceptedChar) return ''
           
           return maskChar === MASK_INPUT_SYMBOL ? char : maskChar + char
       }

        const formattedChar = formatChar(char, maskChar, accept)
```

Reply:
I don't think this is necessary. The format function is the whole codeblock.